### PR TITLE
Database Setup

### PR DESCRIPTION
**Pull request by LancePants97**
## Description
In this branch, I created the database tables including Users, Games, Friendships, and UserGames. In addition, I also created spec tests for all of the models. These include validations and relationship tests.
## Motivation and Context
This is a necessary PR because we will need these database tables in order to build out our API endpoints and create a backend.
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor/File Clean-up
- [X] Tests
- [ ] Documentation
## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] I assigned at least 1 group members to review my PR
## Notes:
Need to update documentation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,6 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+
+/coverage/
```
Comment (CoryBretsch) [inline .gitignore:35]: solid.

**Review by AustinKCodes — APPROVED** [review]

File: db/seeds.rb [hunk 2]
```diff
@@ -7,3 +7,33 @@
...
+user3 = User.create(username: "knob", email: "knob@turing.com", password: "iworshipruby", avatar: "image3_string_goes_here.jpg")
+user4 = User.create(username: "jan", email: "jan@turing.com", password: "imabeast", avatar: "image4_string_goes_here.jpg")
+user5 = User.create(username: "kat", email: "kat@turing.com", password: "bestinstructorever", avatar: "image5_string_goes_here.jpg")
+user6 = User.create(username: "abdul", email: "abdul@turing.com", password: "kingofmod2", avatar: "image6_string_goes_here.jpg")
+user7 = User.create(username: "jeff", email: "jeff@turing.com", password: "iownthisshizz", avatar: "image7_string_goes_here.jpg")
+user8 = User.create(username: "chatgpt", email: "chatgpt@turing.com", password: "robotoverlords", avatar: "image8_string_goes_here.jpg")
+
+friendship1 = Friendship.create(user_id: user1.id, friend_id: user2.id)
+friendship1 = Friendship.create(user_id: user1.id, friend_id: user3.id)
+friendship1 = Friendship.create(user_id: user1.id, friend_id: user4.id)
```
Comment (AustinKCodes) [inline db/seeds.rb]: 2 of these should be renamed

File: app/models/friendship.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
```
Comment (CoryBretsch) [inline app/models/friendship.rb:3]: Nice work - this seems to be an attr_reader method for a users friends.

File: app/models/friendship.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  
+  validates :user_id, uniqueness: { scope: :friend_id }
```
Comment (CoryBretsch) [inline app/models/friendship.rb:5]: very nice work - ensures unique combinations

File: app/models/friendship.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  validate :not_self
```
Comment (CoryBretsch) [inline app/models/friendship.rb:6]: VERY nice work :) - ensures the friend is not the user.

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :user_games
+  has_many :games, through: :user_games
+  has_many :friendships
+  has_many :friends, through: :friendships, source: :friend
+
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
```
Comment (CoryBretsch) [inline app/models/user.rb:7]: dang dude - solid db chops. Ensures we can call friends through from the friend_id

File: spec/models/friendship_spec.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Friendship, type: :model do
+  describe "relationships" do
+    it { should belong_to(:user) }
+    it { should belong_to(:friend) }
+  end
+
+  describe "validations" do
+    it { should validate_uniqueness_of(:user_id).scoped_to(:friend_id) }
```
Comment (CoryBretsch) [inline spec/models/friendship_spec.rb:10]: super solid testing here

File: spec/models/user_spec.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe "relationships" do
+    it { should have_many(:user_games) }
+    it { should have_many(:games).through(:user_games) }
+    it { should have_many(:friendships) }
+    it { should have_many(:friends).through(:friendships).source(:friend) }
+    it { should have_many(:inverse_friendships) }
+    it { should have_many(:inverse_friends).through(:inverse_friendships).source(:user) }
```
Comment (CoryBretsch) [inline spec/models/user_spec.rb:10]: solid testing

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe "relationships" do
+    it { should have_many(:user_games) }
+    it { should have_many(:games).through(:user_games) }
+    it { should have_many(:friendships) }
+    it { should have_many(:friends).through(:friendships).source(:friend) }
```
Comment (CoryBretsch) [inline spec/models/user_spec.rb:8]: this is complex stuff - nice job!

**Review by CoryBretsch — APPROVED** [review]
Nice work Lance!

**Pull request merged**